### PR TITLE
cacheStorage: fix bugs make wpts pass

### PR DESCRIPTION
**Pull request by KhafraDev**
CacheStorage is weird. All tests pass again.

### CODE REVIEW

File: lib/cache/cache.js [hunk 1]
```diff
@@ -146,16 +144,24 @@ class Cache {
     webidl.brandCheck(this, Cache)
     webidl.argumentLengthCheck(arguments, 1, { header: 'Cache.addAll' })
 
-    requests = webidl.converters['sequence<RequestInfo>'](requests)
```
Comment (KhafraDev) [inline lib/cache/cache.js:149]: this is weird: any value is technically allowed in RequestInfo (it gets stringified and then parsed as a URL with the window's location typically), but undefined is very explicitly not allowed here for some reason).

File: lib/cache/cache.js [hunk 2]
```diff
@@ -112,14 +112,12 @@ class Cache {
...
-      const responseObject = new Response(response.body?.source ?? null)
-      const body = responseObject[kState].body
+      const responseObject = new Response(null)
       responseObject[kState] = response
-      responseObject[kState].body = body
       responseObject[kHeaders][kHeadersList] = response.headersList
       responseObject[kHeaders][kGuard] = 'immutable'
 
-      responseList.push(responseObject)
+      responseList.push(responseObject.clone())
```
Comment (KhafraDev) [inline lib/cache/cache.js:120]: this is weird: this whole thing i'm kinda iffy about, somehow this was setting a 0-length uint8array as the source and was locking the stream.

Comment (KhafraDev) [reply lib/cache/cache.js:120]: the cloning itself makes sense, if you matched a Response and then consumed its body, the cached stream would be locked if you matched it again.

**Review by ronag — APPROVED** [review]

File: lib/cache/cache.js (hunk 2 above)
Comment (mcollina) [reply lib/cache/cache.js:120]: How is the body assigned to the new response?

**Review by metcoder95 — APPROVED** [review]

File: lib/cache/cache.js (hunk 2 above)
Comment (KhafraDev) [reply lib/cache/cache.js:120]:
```js
      responseObject[kState] = response
```

`response` here is actually an internal response, not a Response object. It contains the body.

**Pull request merged**